executor/aggfuncs/builder.go

```diff
@@ -676,6 +676,13 @@ func buildLeadLag(aggFuncDesc *aggregation.AggFuncDesc, ordinal int) baseLeadLag
 	defaultExpr = expression.NewNull()
 	if len(aggFuncDesc.Args) == 3 {
 		defaultExpr = aggFuncDesc.Args[2]
+		switch et := defaultExpr.(type) {
+		case *expression.Constant:
+			res, err1 := et.Value.ConvertTo(ctx.GetSessionVars().StmtCtx, aggFuncDesc.RetTp)
+			if err1 == nil {
+				defaultExpr = &expression.Constant{Value: res, RetType: aggFuncDesc.RetTp}
+			}
```

Comment:
what if `err1` is not nil?

Reply:
if err1 not null,  the type conversion is failed, just do nothing and go to the normal process